src/views/statusView.ts

```diff
@@ -406,19 +413,19 @@ export default class StatusView implements vscode.Disposable {
             return;
         }
         const self = this;
-        let index = 0;
-        let progressTicks = ["|", "/", "-", "\\"];
-
         let bar = this.getStatusBar(fileUri);
-        bar.progressTimerId = setInterval(() => {
-            index++;
-            if (index > 3) {
-                index = 0;
-            }
-
-            let progressTick = progressTicks[index];
-            statusBarItem.text = statusText + " " + progressTick;
+        let milliseconds = 0;
+        bar.queryTimer = setInterval(() => {
+            milliseconds += 1000;
+            const timeString = self.formatMillisecondsToTimeString(milliseconds);
+            statusBarItem.text = statusText + " " + timeString;
             self.showStatusBarItem(fileUri, statusBarItem);
-        }, 200);
+        }, 1000);
+    }
+
+    private formatMillisecondsToTimeString(milliseconds: number): string {
```

Comment:
you could do the below for queries taking over an hour, but it's probably not needed.

```TypeScript
private formatMillisecondsToTimeString(milliseconds: number): string {
    const hours = Math.floor(milliseconds / 3600000);
    const minutes = Math.floor((milliseconds % 3600000) / 60000);
    const seconds = Math.floor((milliseconds % 60000) / 1000);

    // Pad minutes/seconds for uniform formatting (e.g. "08:05")
    const mm = minutes.toString().padStart(2, "0");
    const ss = seconds.toString().padStart(2, "0");

    if (hours === 0) {
        // no need to show hours if it's 0
        return `${mm}:${ss}`;
    } else {
        // show hours if we have any
        const hh = hours.toString().padStart(2, "0");
        return `${hh}:${mm}:${ss}`;
    }
}
```